# Individual Project Report

# 1) Data Description:
* Both sets of data were collected by a research group in Computer Science at UBC that is led by Frank Wood to collect data about how people play video games


In [22]:
library(tidyverse)
players <- read_csv("https://raw.githubusercontent.com/lemonkmcampbell-hue/DSCIProject/refs/heads/main/players.csv")
sessions <- read_csv("https://raw.githubusercontent.com/lemonkmcampbell-hue/DSCIProject/refs/heads/main/sessions.csv")
player_summary <- summary(players)
sessions_summary <-summary(sessions)

Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Rows: 1535 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (3): hashedEmail, start_time, end_time
dbl (2): original_start_time, original_end_time

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


## players.csv:
#### Number of Observations and Variables: 
* There are 196 observations and 7 variables
#### Summary Statistics:
|Variable|Minimum|Median|Mean|Maximum|
|--------|-------|------|----|-------|
|age|9.00|19.00|21.14|58.00|
|hours_played|0.00|0.100|5.846|223.100|
#### Overview of Variables:
|Name|Type| Description|
|----|----|------------|
|experience|chr| Level of play experience of the game|
|subscribe|logi| A true or false answer for whether parcipitants have subscribed to the game or not|
|hashedEmail|chr| A sequence of numbers and letters that represents the users email|
|played_hours|dbl|The number of hours played|
|name|chr|Name of User/Participant|
|gender|chr| Gender of User/Participant|
|Age|dbl|Age of User/Participant|
#### Issues in the Data:
* There are some N/A or missing values within age
* May be potential unseen errors in sampling or bias in data collection
* Seperate from the other data set

## sessions.csv:
#### Number of Observations and Variables: 
*There are 1535 observations and 5 variables
#### Summary Statistics:
|Variable|Minimum|Median|Mean|Maximum|
|--------|-------|------|----|-------|
|original_start_time|$1.712*10^{12}$|$1.719*10^{12}$|$1.719*10^{12}$|$1.727*10^{12}$|
|original_end_time|$1.712*10^{12}$|$1.719*10^{12}$|$1.719*10^{12}$|$1.727*10^{12}$|
#### Overview of Variables:
|Name|Type| Description|
|----|----|------------|
|hashedEmail|chr| A sequence of numbers and letters that represents the users email|
|start_time|chr|The date and time the user started playing|
|end_time|chr|The date and time the user finished playing|
|original_start_time|dbl| Timestamp for the start time|
|original_end_time|dbl|Timestamp for the end time|
#### Issues in the Data:
* There are some N/A or missing values within original_end_time
* potentially needs to be tidied as there is multiple spots where end ans start time are mentioned as well as date and time being in the same cell
* May be potential unseen errors in sampling or bias in data collection
* Seperate from the other data set

# 2) Questions:
* **Broad Question**: Which "kinds" of players are most likely to contribute a large amount of data? This is so the team is able to target them in recruiting efforts.
* **Specific Question**: Can a player's level of experience predict the amount of time they play for in the player’s dataset?
* The data will help me address this question as the players dataset includes experience level and hours played. I plan to group based on experience and then calculate summary statistics and create visual models based on those groups. Once I rearrange the data I will also be able to sue the predictiv methods learned in class to achieve a result.

# 3) Exploratory Data Analysis and Visualization:
#### Reading in the data:
* Start by loading in the tidyverse library using the library function
* The data is loaded in during the Data Description section as well as below

In [21]:
players <- read_csv("https://raw.githubusercontent.com/lemonkmcampbell-hue/DSCIProject/refs/heads/main/players.csv")
sessions <- read_csv("https://raw.githubusercontent.com/lemonkmcampbell-hue/DSCIProject/refs/heads/main/sessions.csv")

Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Rows: 1535 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (3): hashedEmail, start_time, end_time
dbl (2): original_start_time, original_end_time

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
